readthedocs/core/mixins.py

```diff
@@ -63,3 +65,20 @@ def dispatch(self, request, *args, **kwargs):
 
     def can_be_cached(self, request):
         return self.cache_response
+
+
+class DeleteViewWithMessage(DeleteView):
+
+    """
+    Delete view that shows a message after deleting an object.
+
+    Refs https://code.djangoproject.com/ticket/21926
+    """
+
+    success_message = None
+
+    def post(self, request, *args, **kwargs):
+        resp = super().post(request, *args, **kwargs)
+        if self.success_message:
+            messages.success(self.request, self.success_message)
+        return resp
```

Comment:
We have some places where we do some validations when deleting an object. If that validation fails, we will still show the success message, that will be confusing.

Reply:
We could check if the response is a redirect, in that case there is a 99% chance that the deletion was successful, since a 200 will mean it will render an error.